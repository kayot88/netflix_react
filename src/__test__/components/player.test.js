import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { Player } from '../../components/player/index.jsx';

describe('<Player/>', () => {
  it('Player rendered with the bunny video', () => {
    const { container, queryByTestId, getByRole } = render(
      <Player>
        <Player.Button />
        <Player.Video src="/videos/bunny.mp4" />
      </Player>
    );
    expect(queryByTestId('player')).toBeFalsy();

    fireEvent.click(screen.getByText('Play'));
    expect(queryByTestId('player')).toBeTruthy();

    screen.getByRole('close');
    fireEvent.click(screen.getByRole('close'));
    expect(queryByTestId('player')).toBeFalsy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
