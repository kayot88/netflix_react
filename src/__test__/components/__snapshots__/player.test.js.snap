// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player/> Player rendered with the bunny video 1`] = `
<div
  class="player__Container-sc-1c49yiu-0"
>
  <button
    class="player__Button-sc-1c49yiu-1 cyjoXP"
  >
    Play
  </button>
</div>
`;
